src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordDeconstruct.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Reflection;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SynthesizedRecordDeconstruct : SynthesizedInstanceMethodSymbol
+    {
+        private readonly int _memberOffset;
+        private readonly ImmutableArray<PropertySymbol> _properties;
+
+        public SynthesizedRecordDeconstruct(
+            SourceMemberContainerTypeSymbol containingType,
+            ImmutableArray<ParameterSymbol> ctorParameters,
+            ImmutableArray<PropertySymbol> properties,
+            int memberOffset)
+        {
+            _memberOffset = memberOffset;
+            _properties = properties;
```

Comment:
nit: Consider asserting that all the properties have a getter

Reply:
Done